code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -738,11 +738,9 @@
 	M.do_jitter_animation(10)
 	addtimer(CALLBACK(M, /mob/living/carbon.proc/do_jitter_animation, 10), 40) //jitter immediately, then again after 4 and 8 seconds
 	addtimer(CALLBACK(M, /mob/living/carbon.proc/do_jitter_animation, 10), 80)
-	addtimer(CALLBACK(src, .proc/do_strange_revive,M,reac_volume,amount_to_revive), 79) //timing is everything!
 	..()
-
-/datum/reagent/medicine/strange_reagent/proc/do_strange_revive(mob/living/M,reac_volume,revive_requirement) //we store revive_requirement because if we calculate it now we might have changed our damage numbers.
-	var/excess_healing = 5*(reac_volume-revive_requirement) //excess reagent will heal blood and organs across the board
+	sleep(79)
```

Comment:
Wouldn't it be better to just set a GLOBAL_PROC timer?

Reply:
i think ongoing procs keep the object from being deleted, but i'm not 100% sure

Reply:
why not actually move this proc to mob/ and just call it on the mob as advertised?

Reply:
I was asked by cobby to change it back to a sleep over concerns of adding another proc to /mob/living/. I'm open to using GLOBAL_PROC or a callback to mob.

Reply:
cobby why

Reply:
this is awful but technically a component would serve well here

Reply:
if we can put it on the revive proc itself as an argument that might be better than adding a new proc on all mobs for the sake of a magic chem that doesn't even get made too often depending on who's playing.

Reply:
yeah just change the callback proc from being on the reagent to being on the mob problem solved.

Reply:
I've added the callback to revive.

Reply:
Oh and I changed the log, because we loose the reference to the reagent

Reply:
I personally believe that junking up giant procs is a great sin, and all maintainers will pay for their support of boolean parameters over a global proc

Reply:
scared?

Reply:
I still think a "strange_revival" component that holds the proc should work

Reply:
component component

Reply:
holder factory holder